src/jit/hwintrinsicxarch.cpp

```diff
@@ -716,6 +723,70 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
             }
             break;
 
+        case NI_SSE2_ConvertScalarToVector128Double:
+        {
+            assert(sig->numArgs == 2);
+            assert(getBaseTypeOfSIMDType(sig->retTypeSigClass) == TYP_DOUBLE);
+
+            argList = info.compCompHnd->getArgNext(sig->args);
+            CorInfoType corType =
+                strip(info.compCompHnd->getArgType(sig, argList, &argClass)); // type of the second argument
+
+            baseType = JITtype2varType(corType);
+
+#ifdef _TARGET_X86_
+            if (varTypeIsLong(JITtype2varType(corType)))
```

Comment:
This case should already be handled by `impHWIntrinsic`

Reply:
> This case should already be handled by impHWIntrinsic

Currently, `impHWIntrinsic` only handle `r64` for return types. Maybe we can make the change latter.

Reply:
Do we have a bug tracking this yet?

Do we also have any mechanism for determining when the intrinsic takes/returns a 64-bit value but also works on 32-bit machines (there is at least a couple of them)?

Reply:
> Do we have a bug tracking this yet?

I do not think this is a bug. We can introduce more flags to hnit the framework that the intrinsic may have an `r64` oprand on first/second/.. position.

Reply:
Let me make this change after this PR merged.